pkg/ccl/backupccl/restore_job.go

```diff
@@ -1805,24 +1805,41 @@ func insertStats(
 		return nil
 	}
 
-	if latestStats == nil {
-		return nil
-	}
+	// We could be restoring hundreds of tables, so insert the new stats in
+	// batches instead of all in a single, long-running txn. This prevents intent
+	// buildup in the face of txn retries.
+	batchSize := 10
```

Comment:
I could see moving the `10` out to a package-level constant: `restoreStatsInsertBatchSize`.  If 10 was an arbitrary choice, might be good to say that in the comment as well.